# Welcome! Take a look at the code below, starting with #1. 
# What type of error does this produce and why?

# ERROR 1:
# result = 10 / 0  
# print("This will not be printed due to the crash.")
# ZeroDivisionError - cannot divide zero

# ERROR 2:
# result = "10" + 5
# print(result) 
# print("This will not be printed due to the crash.")
# TypeError - the two types of data available are not compatable to concatenate

# ERROR 3:
# print(unknown_variable) 
# print("This will not be printed due to the crash.")
# NameError - the compiler doesnt know what this name references

# ERROR 4:
# lst = [1, 2, 3]
# print(lst[5])  
# print("This will not be printed due to the crash.")
# IndexError - we are trying to access some information in a list that is unavailable / doesnt exist

# ERROR 5:
# d = {"name": "Dylan"}
# print(d["age"]) 
# print("This will not be printed due to the crash.")
# KeyError - the key in a dictionary is unaccessable / doesnt exist

# ERROR 6: 
# num = int("Hello")  
# print("This will not be printed due to the crash.")
# ValueError - program expects one type of data and has recieved a different type

# ERROR 7:
# console.log("Hello Console!")
# print("This will not be printed due to the crash.")
# SyntaxError (^^ above is not a syntax error) - we have written something incorrectly

